### PR TITLE
Add some Vale styles

### PR DESCRIPTION
**Pull request by stichbury**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description
This PR adds styles to the Vale linter

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: .github/styles/Kedro/ukspelling.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    - catalog
+    - Catalog
+    - dialog
+    - log
+    - maize
+    - prize
+    - seize
+    - size
+    - tailor
+
```
Comment (ankatiyar) [inline .github/styles/Kedro/ukspelling.yml:25]: Also add "DataCatalog" to the list of exceptions?

File: .github/styles/Kedro/words.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  name: replace
+swap:
+  # bad: good
+  'acknowledgement': 'acknowledgment'
+  'auto-complete': 'autocomplete'
+  'a number of': 'few|several|many'
+  'and/or': 'and|or|either or'
+  'back end': 'backend'
+  'bear in mind': 'keep in mind'
+  'culprit': 'cause'
```
Comment (deepyaman) [inline .github/styles/Kedro/words.yml:15]:
```suggestion
  'culprit': 'cause'
  'data set': 'dataset'
```
:)

(Also, would apply to the "web site" rule, too--do you need to create a separate rule for the plural?)

**Comment by stichbury** [comment]
> LGTM! My only concern would be if get a lot of warnings (not errors), to the point people end up ignoring them, because there are a lot of rules. I like it, but I'd be curious to see how many issues this come out with. :)

This is my worry too, which is why I want to try it out over the next sprint or two. I've already dropped the level from error->warning for most things and removed some of the less "wordy" words from `toowordy.yml`. If it's getting too annoying I'll revise it.

**Review by astrojuanlu — APPROVED** [review]
The rules look great! Can't wait to see the automatic comments on PRs 😄

**Review by ankatiyar — APPROVED** [review]
LGTM 🥇

**Review by DimedS — COMMENTED** [review]
Great styles! With that linter, we'll all become professional English writers! 👍

**Pull request merged**